app/services/logging/logger.js

```diff
@@ -1,17 +1,143 @@
-'use strict';
+"use strict";
+
+const path = require("path");
+
+class LoggerTransports {
+  static CONSOLE = "console";
+  static FILE = "file";
+  static ROLLING = "rolling";
+}
 
 module.exports = class Logger {
-  constructor() {
+  constructor({ winston, loggerConfig }) {
+    this.winston = winston;
+    this.loggerConfig = loggerConfig;
   }
 
-  log(message, ...optionalParams) {
-    // TODO: Use moment or similar to format the date
-    // TODO: Replace console.log() with a logging tool
-    message = `${new Date().toLocaleString("en-GB")}: ${message}`
+  init() {
```

Comment:
This method is doing too much; you can split the implementation into multiple methods to make testing easier / more concise - in OO these sub-methods would be "private", but as JS has no notion of access modifiers, this makes unit testing of each sub-method very easy and independent.

Generally, if, else, else if constructs are a bit of a code smell and make code harder to read and maintain in the long run. Without really digging into it, I'd generally expect each condition (if or else section) to be its own, independent method. In this case, you'd make more use of the `return` statement to control the flow.

Reply:
Agreed. I'm going to move these into separate transport classes each which checks to see if their handle has been mentioned in the config.